### PR TITLE
Minor refactorings to allow low-level shell db explore tool

### PR DESCRIPTION
**Pull request by tinwelint**

### CODE REVIEW

File: community/shell/src/main/java/org/neo4j/shell/impl/AbstractAppServer.java [hunk 1]
```diff
@@ -52,21 +52,30 @@
...
+            for ( App app : Service.load( App.class ) )
+            {
+                addApp( app );
+            }
         }
-	}
+    }
 
-    private void addApp( App app )
+    protected void addApp( App app )
```
Comment (davidegrohmann) [inline community/shell/src/main/java/org/neo4j/shell/impl/AbstractAppServer.java:78]: why is this needed?

Comment (tinwelint) [inline community/shell/src/main/java/org/neo4j/shell/impl/AbstractAppServer.java:78]: For other shell app servers to add their own `App` instances. Like if I'm writing a custom shell tool that has nothing to do with the normal high-level neo4j shell apps I'm going to need this.
